app/src/main/java/kr/ac/konkuk/gdsc/plantory/presentation/plant/AddPlantFragment.kt

```diff
@@ -57,31 +67,61 @@ class AddPlantFragment : BindingFragment<FragmentAddPlantBinding>(R.layout.fragm
     private val getContent =
         registerForActivityResult(ActivityResultContracts.GetContent()) { uri: Uri? ->
             if (uri != null) {
-                val newPlantInfo = viewModel.plantInfo.value.copy(image = uri)
-                viewModel.updatePlantInfo(newPlantInfo)
+                viewModel.updatePlantImage(uri)
+                updateRequestBody()
             }
         }
 
+    private fun initRegisterButtonClickListener() {
+        binding.btnAddplantUpload.setOnSingleClickListener {
+            viewModel.viewModelScope.launch {
+                viewModel.postRegisterPlant()
+            }
+        }
+    }
+
+    private fun setPostRegisterPlantStateObserver() {
+        viewModel.postRegisterPlantState.flowWithLifecycle(viewLifeCycle).onEach { state ->
+            when (state) {
+                is UiState.Success -> {
+                    Timber.d("Success : Register ")
+                    parentFragmentManager.popBackStack()
+                }
+
+                is UiState.Failure -> {
+                    Timber.d("Failure : ${state.msg}")
+                }
+
+                is UiState.Empty -> {
+                }
+
+                is UiState.Loading -> {
+                }
```

Comment:
```suggestion
                is UiState.Empty -> Unit

                is UiState.Loading -> Unit
```

이런식으로도 적을 수 있다는 사실!! (권장은 아님, 그냥 이렇게도 적을 수 있다구요)